docs/data/sdks/next-gen-sdks.md

```diff
@@ -0,0 +1,186 @@
+
+Amplitude puts an SDK into maintenance status and under the "**Maintenance SDKs**" section in the left bar if another same-platform next-gen SDK is available, for example, Browser, Android, React Native, and Node.js. Maintenance SDKs only receives bug fixes until deprecation. It's strongly recommended to adopt next-gen SDKs to take advantage of their benefits. To migrate from a maintenance SDK, refer to the migration guide under each next-gen SDK documentation. 
+
+!!!note 
+    There are still some SDKs that aren't nex-gen while also not in maintenance, for example Flutter, Java, and iOS. 
+
+Refer to the [SDK status table](./#sdk-status-table) to check whether a SDK is next-gen or in maintenance. 
+
+## Architecture
+
+Next-gen SDKs share the same architecture and interfaces across platform. Refer to each SDK full documentation for more details. 
+
+![Next-gen Architecture](/../assets/images/data-next-gen-sdk-architecture.png)
+
+### Client
+
+Amplitude Client stores configurations and is also the entry point for developers to track events. 
+
+Before instrument, you must initialize a SDK using the API key for your Amplitude project, which creates an Amplitude Client. 
+
+```python 
+# Next-gen Python SDK example: initialize a Client
+
+from amplitude import Amplitude
+
+amplitude = Amplitude('API_KEY')
+```
+
+### Configuration
+
+Configuration includes the configurable options to configure Client behavior. 
+
+You can pass a Configuration when Client initialization.
+
+```python
+# Next-gen Python SDK example: initialize a Client with a Configuration
+
+from amplitude import Amplitude
+from amplitude import Config
+
+amplitude = Amplitude("API_KEY", 
+                      configuration=Config(
+                            flush_queue_size=100, # flush queued events when there are 100 or more
+                            flush_interval_millis=20000 # flush queued events every 20 seconds
+                    )) 
+```
+
+You can also get access to the Configuration of your Client after initialization.
+
+```python
+Next-gen Python SDK example: get access to the Configuration of your Client
+
+from amplitude import Amplitude
+
+client = Amplitude('API_KEY')
+# flush queued events when there are 100 or more
+client.configuration.flush_queue_size = 100
+# flush queued events every 20 seconds
+client.configuration.flush_interval_millis = 20000
+```
+
+### Storage provider
+
+A storage provider holds events in a designated storage buffer. The default storage provider only queues events in memory. To prevent data lose, you can customize your own storage provider to persist events, for example to a disk, in case of unexpected shutdown. This way, you can ensure that you can recover un-sent events even in the event of a system failure. 
+
+You can configure the storage provider in the Configuration of your Client. Refer to the "Configuration Options" section of a specific next-gen SDK full documentation for more details.
+
```

Comment:
StorageProvider code snippet?

Reply:
Added a StorageProvider Browser SDK example, but left the methods blank. 
If there is a good example, we could fill it up later

```class MyStorage<T> {
  isEnabled(): Promise<boolean> {
    // Whether the storage is enabled
  }

  get(key: string): Promise<T | undefined> {
    // get the value associated with the given key in the storage
  }

  getRaw(key: string): Promise<string | undefined> {
    // get the string associated with the given key in the storage
  }

  set(key: string, value: T): Promise<void> {
    // set an item in the storage by key
  }

  remove(key: string): Promise<void> {
    // remove an item in the storage by key
  }

  reset(): Promise<void> {
    // clear the storage
}
```